lib/internal/queue.js

```diff
@@ -45,16 +47,14 @@ export default function queue(worker, concurrency, payload) {
     }
 
     function _next(tasks) {
-        return function(){
+        return rest(function(args){
             workers -= 1;
 
-            var removed = false;
-            var args = arguments;
             arrayEach(tasks, function (task) {
-                arrayEach(workersList, function (worker, index) {
-                    if (worker === task && !removed) {
+                arraySome(workersList, function (worker, index) {
```

Comment:
I'm pretty sure we only execute each task once, because we remove them as they are processed.  I could be missing some detail though.

I think using `some` for short circuiting is a bit hacky.  It also means we don't have to import `arraySome`.